### PR TITLE
tests: fix compact byte detection

### PR DESCRIPTION
**Pull request by BusyJay**
test_compact_lock_cf often fails on my machine.

@siddontang @zhangjinpeng1987 PTAL

### CODE REVIEW

File: src/util/file.rs [hunk 1]
```diff
@@ -105,7 +104,8 @@ mod test {
         let perm_file = dir_path.join("perm_file");
         {
             let f = OpenOptions::new().write(true).create_new(true).open(&perm_file).unwrap();
-            let perm = Permissions::from_mode(0o400);
+            let mut perm = f.metadata().unwrap().permissions();
+            perm.set_readonly(true);
```
Comment (siddontang) [inline src/util/file.rs]: any concern to not using 400 here?

Comment (BusyJay) [inline src/util/file.rs]: Not working on Windows.

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**